tests/output.rs

```diff
@@ -1,228 +1,118 @@
+    let mut path: PathBuf = root_dir.into();
+    path.push("tests/fixtures");
+    path.push(fixture_dir_name);
+    path
 }
 
-impl From<((usize, usize), (usize, usize))> for Range {
-    fn from(value: ((usize, usize), (usize, usize))) -> Self {
-        Self {
-            start: value.0.into(),
-            end: value.1.into(),
+fn parse_command_and_output(command_and_output: &str) -> CommandAndOutput {
+    let mut lines = command_and_output.split("\n").collect::<Vec<_>>();
+    if !lines.is_empty() {
+        if regex!(r#"^\s*$"#).is_match(&lines[0]) {
+            lines.remove(0);
         }
     }
+    let command_line = lines.remove(0);
+    let indent = regex!(r#"^\s*"#).find(&command_line).unwrap().as_str();
+    let command_line_args = parse_command_line(strip_indent(&command_line, indent));
+    if !lines.is_empty() {
+        if regex!(r#"^\s*$"#).is_match(&lines[lines.len() - 1]) {
+            lines.pop();
+        }
+    }
+    let output: String = lines
+        .into_iter()
+        .map(|line| {
+            assert!(line.starts_with(indent));
+            format!("{}\n", strip_indent(line, indent))
+        })
+        .collect();
+    CommandAndOutput {
+        command_line_args,
+        output,
+    }
 }
 
-struct ExpectedMatch {
-    pub relative_file_path: String,
-    pub lines: Vec<String>,
-    pub range: Range,
+struct CommandAndOutput {
+    command_line_args: Vec<String>,
+    output: String,
 }
 
-const MATCH_OUTPUT_LINE_REGEX_STR: &'static str = r#"(^|\n).+:\d+:"#;
-
-fn predicate_from_expected_matches(
-    expected_matches: &[ExpectedMatch],
-    output_mode: OutputMode,
-) -> BoxPredicate<str> {
-    expected_matches.into_iter().fold(
-        BoxPredicate::new(
-            predicate::str::is_match(MATCH_OUTPUT_LINE_REGEX_STR)
-                .unwrap()
-                .count(output_mode.get_expected_total_number_of_match_lines(expected_matches)),
-        ),
-        |predicate, expected_match| {
-            BoxPredicate::new(
-                predicate.and(predicate_from_expected_match(expected_match, output_mode)),
-            )
-        },
-    )
+fn strip_indent<'line>(line: &'line str, indent: &str) -> &'line str {
+    &line[indent.len()..]
 }
 
-fn predicate_from_expected_match(
-    expected_match: &ExpectedMatch,
-    output_mode: OutputMode,
-) -> BoxPredicate<str> {
-    BoxPredicate::new(predicate::str::contains(
-        output_mode.expected_output_text(expected_match),
-    ))
+fn parse_command_line(command_line: &str) -> Vec<String> {
+    assert!(command_line.starts_with("$"));
+    shlex::split(&command_line[1..]).unwrap()
 }
 
-fn assert_expected_matches<TArg: AsRef<OsStr>>(
-    args: impl IntoIterator<Item = TArg>,
-    current_dir: impl AsRef<Path>,
-    expected_matches: &[ExpectedMatch],
-    output_mode: OutputMode,
-) {
-    Command::cargo_bin("tree-sitter-grep")
+fn assert_sorted_output(fixture_dir_name: &str, command_and_output: &str) {
```

Comment:
The way you have tests written currently (essentially 1-statement per test) it doesn't matter, but I think it's generally worth making assert functions macros, so that you get the panic reported for a line that's actually in a test.

Reply:
Also maybe worth considering using something like https://crates.io/crates/pretty_assertions to get nice diffs in the assertions.

Reply:
> I think it's generally worth making assert functions macros, so that you get the panic reported for a line that's actually in a test

Ah interesting. I wonder if `#[track_caller]` would accomplish the same thing (that's what I saw being used in `moxie` for the stuff related to "what's the current source-code location", I think it basically says "if I get asked what my source-code location is (eg for backtraces?) I'll lie and tell them the location of whoever called me instead")?

> Also maybe worth considering using something like https://crates.io/crates/pretty_assertions to get nice diffs in the assertions

Ya that would be nice, and there have been a couple places I might've reached for [`speculoos`](https://crates.io/crates/speculoos)-style assertions (which I've been liking for the Typescript compiler in Rust tests), but I'm not sure I totally understand how/where to integrate that with these existing assertions eg `predicates` that that guide was using